src/precice/config/ParticipantConfiguration.cpp

```diff
@@ -559,5 +561,52 @@ void ParticipantConfiguration::finishParticipantConfiguration(
   _isMasterDefined = false; // to not mess up with previous participant
 }
 
+void ParticipantConfiguration::checkIllDefinedMappings(
+    const mapping::MappingConfiguration::ConfiguredMapping &mapping,
+    const impl::PtrParticipant &                            participant)
+{
+  PRECICE_TRACE();
+  using ConfMapping = mapping::MappingConfiguration::ConfiguredMapping;
+
+  for (const ConfMapping &configuredMapping : _mappingConfig->mappings()) {
+    bool sameToMesh   = mapping.toMesh->getName() == configuredMapping.toMesh->getName();
+    bool sameFromMesh = mapping.fromMesh->getName() == configuredMapping.fromMesh->getName();
+    if (sameToMesh && sameFromMesh) {
+      // It's really the same mapping, duplicated mappings are already checked for in MappingConfiguration
+      return;
+    }
```

Comment:
Shouldn't this be an assertion?

Reply:
No, I altered the comment to make this easier to understand.
"It's really the same mapping, not a duplicated one. Those are already checked for in MappingConfiguration."
Clear?

Reply:
Got it! Yes its clear